sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -193,7 +193,16 @@ class SessionCatalog(
   def getDatabaseMetadata(db: String): CatalogDatabase = {
     val dbName = formatDatabaseName(db)
     requireDbExists(dbName)
-    externalCatalog.getDatabase(dbName)
+    val database = externalCatalog.getDatabase(dbName)
+
+    // default database's location always use the warehouse path,
+    // and since the location of database stored in metastore is qualified,
+    // here we also make qualify for warehouse location
```

Comment:
How about 

> // The default database's location always uses the warehouse path.
// Since the location of database stored in metastore is qualified,
// we also make the warehouse location qualified.